Lib/shutil.py

```diff
@@ -813,6 +813,11 @@ def move(src, dst, copy_function=copy2):
             if _destinsrc(src, dst):
                 raise Error("Cannot move a directory '%s' into itself"
                             " '%s'." % (src, dst))
+            if ((not os.access(src, os.W_OK) and os.listdir(src))
+                    or _is_immutable(src)):
+                raise PermissionError("Cannot move the non-empty directory "
+                                      "'%s': Lacking write permission to '%s'."
+                                      % (src, src))
```

Comment:
1. Is this `src` to `src` ? Shouldn't the  verification in line 816 be on `dst`?

2. A general question - if this applicable only to macos, should we need OS platform check? - I am not certain on this.

Reply:
1. `src` can't be moved to `dst` if: (`src` is not empty) && (user lacks write permission to `src`). So we're only checking the status of `src` here. 
2. Good catch, the immutability check is applicable to all platforms, but the empty/permissions check is only applicable to darwin. I've added a check.